include/swift/SIL/SILOpenedArchetypesTracker.h

```diff
@@ -44,12 +36,6 @@ class SILOpenedArchetypesTracker : public DeleteNotificationHandler {
   SILOpenedArchetypesTracker(SILOpenedArchetypesTracker &Tracker)
       : SILOpenedArchetypesTracker(Tracker.F, Tracker) {}
```

Comment:
It looks cleaner, but I don't like the use of new/delete. Could you try to move the creation of the cloner instance into `CheckedCastBrJumpThreading::optimizeFunction`, which calls both functions that set the `Cloner` value? You could allocate the `Cloner` there on stack and just pass its address or something like that.  Or may be the Cloner instance can be created without the new/delete as part of the `Edit` constructor and becomes a part of the Edit structure, because that one sets the CCBlock which actually may need a cloner later?